core/src/main/java/com/linecorp/armeria/internal/client/ClientRequestContextExtension.java

```diff
@@ -73,4 +75,15 @@ public interface ClientRequestContextExtension extends ClientRequestContext, Req
      * with default values on every request.
      */
     HttpHeaders internalRequestHeaders();
+
+    /**
+     * Sets the {@link Consumer} which will be invoked when the response is closed.
+     */
+    void setClosingResponseTask(Consumer<@Nullable Throwable> closingResponseTask);
```

Comment:
What do you think of adding the word `websocket` in the naming so that we don't have to look up what this does every time?

i.e. `webSocketResponseCloseTask` or `onWebSocketCloseFrame`